### ISSUE TITLE
tag_no_case is matching any input of the same length

### ISSUE THREAD

**Issue by jtdowney** [Geal/nom#387]
When starting a new project with 2.0 I noticed that `tag_no_case` was matching any characters, not just the ones specified. Here is an example:

```rust
#[macro_use]
extern crate nom;

use nom::IResult::*;

named!(test, tag_no_case!("abcd"));

fn main() {
    let r = test(&b"aBCdefgh"[..]);
    assert_eq!(r, Done(&b"efgh"[..], &b"aBCd"[..]));

    let r = test(&b"dogbone"[..]);
    assert_eq!(r, Done(&b"one"[..], &b"dogb"[..]));
}
```

I would expect the first one to match as it does but the second to be `IResult::Error(Tag)`.

**Comment by jtdowney** [comment]
My bitwise math is a bit rusty (no pun) but I think https://github.com/Geal/nom/blob/844edaff8a21ee2e3662ef3f3ce3919547f94ef9/src/traits.rs#L331 should be a bitwise `OR` with `0b00100000`. Right now it is an `AND` with `0b01000000`.

**Comment by Geal** [comment]
this is awful, I should have caught that before the release :(

I just fixed it in 1f6202b, thanks for the report, I'll release a 2.0.1 soon.
